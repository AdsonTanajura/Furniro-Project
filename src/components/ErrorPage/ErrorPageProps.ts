interface ErrorPageProps {
  errorMenssage?: string;
}

export default ErrorPageProps;
