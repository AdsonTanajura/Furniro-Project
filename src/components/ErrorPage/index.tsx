import React from 'react';
import ErrorPageProps from './ErrorPageProps';
import { ErrorPageContainer } from './styles';

const ErrorPage = ({
  errorMenssage = 'Sorry about what happened',
}: ErrorPageProps) => {
  return (
    <ErrorPageContainer>
      <h2>We had an error</h2>
      <span>ğŸ™€</span>
      <p>{!errorMenssage ? 'Sorry about what happened' : errorMenssage}</p>
    </ErrorPageContainer>
  );
};

export default ErrorPage;
