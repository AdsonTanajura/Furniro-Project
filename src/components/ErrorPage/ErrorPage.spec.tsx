import { render } from '@testing-library/react';
import ErrorPage from '.';

describe('<ErrorPage />', () => {
  const defultMenssage = 'Sorry about what happened';
  it('should render the error container', () => {
    const { container } = render(<ErrorPage />);

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display a default message when no errorMenssage is provided', () => {
    const { getAllByRole, getAllByText, container } = render(<ErrorPage />);

    const titles = getAllByRole('heading', { level: 2 });
    const emojis = getAllByText('ðŸ™€');
    const errorMensages = getAllByText(defultMenssage);

    expect(titles.length).toBeGreaterThan(0);
    titles.forEach((title) => expect(title).toBeInTheDocument());
    titles.forEach((title) => expect(title).toBeVisible());

    expect(errorMensages.length).toBeGreaterThan(0);
    errorMensages.forEach((errorMensage) => expect(errorMensage).toBeVisible());
    errorMensages.forEach((errorMensage) =>
      expect(errorMensage).toBeInTheDocument()
    );

    expect(emojis.length).toBeGreaterThan(0);
    emojis.forEach((emoji) => expect(emoji).toBeVisible());
    emojis.forEach((emoji) => expect(emoji).toBeInTheDocument());

    expect(container.firstChild).toMatchSnapshot();
  });

  it('the error message passed in by parameter must be visible to users', () => {
    const message = 'my bed error';
    const { getAllByText, getAllByRole, container } = render(
      <ErrorPage errorMenssage={message} />
    );

    const emojis = getAllByText('ðŸ™€');
    const titles = getAllByRole('heading', { level: 2 });
    const errorMensages = getAllByText(message);

    expect(titles.length).toBeGreaterThan(0);
    titles.forEach((title) => expect(title).toBeVisible());
    titles.forEach((title) => expect(title).toBeInTheDocument());

    expect(emojis.length).toBeGreaterThan(0);
    emojis.forEach((emoji) => expect(emoji).toBeVisible());
    emojis.forEach((emoji) => expect(emoji).toBeInTheDocument());

    expect(errorMensages.length).toBeGreaterThan(0);
    errorMensages.forEach((errorMensage) => expect(errorMensage).toBeVisible());
    errorMensages.forEach((errorMensage) =>
      expect(errorMensage).toBeInTheDocument()
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should handle an empty or undefined errorMenssage', () => {
    const { getAllByAltText } = render(<ErrorPage errorMenssage={''} />);
    const emojis = getAllByAltText('ðŸ™€');
    const errorMensages = getAllByAltText(defultMenssage);

    expect(emojis.length).toBeGreaterThan(0);
    emojis.forEach((emoji) => expect(emoji).toBeVisible());
    emojis.forEach((emoji) => expect(emoji).toBeInTheDocument());

    expect(errorMensages.length).toBeGreaterThan(0);
    errorMensages.forEach((errorMenssage) =>
      expect(errorMenssage).toBeVisible()
    );
    errorMensages.forEach((errorMenssage) =>
      expect(errorMenssage).toBeInTheDocument()
    );
  });
});
