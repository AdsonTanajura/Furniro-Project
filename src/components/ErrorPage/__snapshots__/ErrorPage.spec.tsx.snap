// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> should display a default message when no errorMenssage is provided 1`] = `
<div
  class="sc-VHjGu bvRlJb"
>
  <h2>
    We had an error
  </h2>
  <span>
    ðŸ™€
  </span>
  <p>
    Sorry about what happened
  </p>
</div>
`;

exports[`<ErrorPage /> should render the error container 1`] = `
<div
  class="sc-VHjGu bvRlJb"
>
  <h2>
    We had an error
  </h2>
  <span>
    ðŸ™€
  </span>
  <p>
    Sorry about what happened
  </p>
</div>
`;

exports[`<ErrorPage /> the error message passed in by parameter must be visible to users 1`] = `
<div
  class="sc-VHjGu bvRlJb"
>
  <h2>
    We had an error
  </h2>
  <span>
    ðŸ™€
  </span>
  <p>
    my bed error
  </p>
</div>
`;
